### PR TITLE
Signup reskin: Update UI for domain search results

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Changes proposed in this Pull Request

This PR is towards implementing the reskinned signup experiment. For more context on the experiment, read pbxNRc-IO-p2.

NOTE:

This PR implements only the UI changes for the domain search results. Other UI changes are implemented in subsequent PRs.

Here's the [Figma file](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1114%3A536) for reference.

For the record:
Parent PR: https://github.com/Automattic/wp-calypso/pull/51504 and root PR: https://github.com/Automattic/wp-calypso/pull/50512

**With search results:**

Screenshot 2021-03-31 at 5 21 48 PM

**Search results with on-hover effect:**

Screenshot 2021-03-31 at 5 21 58 PM

**Search results with the ".com" tld filter applied:**

Screenshot 2021-03-31 at 5 22 12 PM

#### Testing instructions

* Assign yourself to treatment from Abacus, and visit /start.
*  Proceed to the domains step and verify that the UI matches the [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1114%3A536). 
* Start over and verify that control group experience is unaffected

### CODE REVIEW

File: client/components/domains/domain-product-price/index.jsx [hunk 1]
```diff
@@ -87,11 +87,50 @@ class DomainProductPrice extends React.Component {
...
+
+		const message = translate( '{{span}}Free for the first year{{/span}}', {
+			components: { span: <span className="domain-product-price__free-price" /> },
+		} );
+
+		return domainPriceElement( message );
+	}
+
+	renderReskinDomainPrice() {
+		const priceText = this.props.translate( '%(cost)s/year', {
```
Comment (a8ci18n) [inline client/components/domains/domain-product-price/index.jsx:109]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( '%(cost)s /year' )` *ES Score: 7*

**Comment by dzver** [comment]
So, I did a round of testing on this PR and found a couple of things that might be issues.

1. I see this on Mobile in treatment:
Screenshot 2021-04-01 at 12 42 51

It works well in control.

2. When I run `yarn install`, I see this
Screenshot 2021-04-01 at 12 45 22

**Comment by niranjan-uma-shankar** [comment]
> I see this on Mobile in treatment:

I should've mentioned this in the testing instructions but I'm working on mobile and tablet UIs in a separate PR https://github.com/Automattic/wp-calypso/pull/51611


> When I run yarn install, I see this

It doesn't seem to be related but I can take a look more deeply

**Comment by klimeryk** [comment]
Tested the control/current version first (as I need to figure out how to "assign myself to treatment from Abascus" (don't worry, I can MGS/FG ;)), but so far, so good. Only one minor nitpick is that the "Use a domain I own" is now broken into two lines, despite having enough space:

Screenshot 2021-04-01 at 17 30 16

On production:
Screenshot 2021-04-01 at 17 30 11

**Review by dzver — APPROVED** [review]
LGTM

**Comment by niranjan-uma-shankar** [comment]
> Only one minor nitpick is that the "Use a domain I own" is now broken into two lines, despite having enough space:

Yup, I noticed and fixed it in a different PR https://github.com/Automattic/wp-calypso/pull/51611. Thanks for testing and reporting it!

**Pull request merged**